### PR TITLE
Random whiteship selection

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Breaks shit if the deprecated map pr is merged before/after without me fixing it so only merge one of those

## Why It's Good For The Game

yea

## Changelog
:cl:
add: Whiteship is randomly chosen
/:cl:

### CODE REVIEW

**Comment by EdgeLordExe** [comment]
@TiviPlus Pubby whiteship and delta white ship are both awful please at least make it so pubby one can move.

File: code/modules/shuttle/shuttle.dm [hunk 1]
```diff
@@ -246,6 +246,27 @@
 		reserved_area = null
 	return ..()
 
+/obj/docking_port/stationary/picked
+	var/list/shuttlekeys
```
Comment (bobbahbrown) [inline code/modules/shuttle/shuttle.dm:250]: This comment (or lack thereof) does not meet our [documentation standards](https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md#documenting-a-variabledefine), please update this.

**Comment by msgerbs** [comment]
Do all whiteship docks on all maps fix all other maps whiteships? There are significant size differences here.

**Comment by Dennok** [comment]
> Do all whiteship docks on all maps fit all other maps whiteships? There are significant size differences here.

As i memorize, some no. But you can place custom dock where you want.

File: code/modules/shuttle/shuttle.dm [hunk 2]
```diff
@@ -246,6 +245,28 @@
 		reserved_area = null
 	return ..()
 
+/obj/docking_port/stationary/picked
```
Comment (msgerbs) [inline code/modules/shuttle/shuttle.dm:248]: Holds?

File: code/modules/shuttle/shuttle.dm [hunk 3]
```diff
@@ -246,6 +245,28 @@
...
+	///Hods a list of map name strings for the port to pick from
```
Comment (msgerbs) [inline code/modules/shuttle/shuttle.dm:249]: Holds?

File: code/modules/shuttle/shuttle.dm [hunk 4]
```diff
@@ -246,6 +245,28 @@
...
+
+/obj/docking_port/stationary/picked/whiteship
+	name = "Deep Space"
+	id = "whiteship_away"
+	dheight = 0
+	dir = 2
+	dwidth = 11
+	height = 22
+	width = 35
+	shuttlekeys = list("whiteship_meta", "whiteship_pubby", "whiteship_box", "whiteship_cere", "whiteship_kilo", "whiteship_donut", "whiteship_delta")
```
Comment (msgerbs) [inline code/modules/shuttle/shuttle.dm:268]: What happens if the picked key doesn't actually exist? Say somebody removes one of these ships but doesn't know they need to update this var?

Comment (TiviPlus) [reply code/modules/shuttle/shuttle.dm:268]: It would have the exact same failcase as the json based system the whiteships use now

Comment (msgerbs) [reply code/modules/shuttle/shuttle.dm:268]: I don't know what that failcase is, but good enough for me.

**Review by Timberpoes — APPROVED** [review]
Tivi's assured me this is tested. If whiteships are incompatible with stations, we'll fix any issues as they crop up.

PR seems well received, so let's greenlight it.

image

As long as you tell me WHICH FREAKIN' PR YOU'RE WAFFLING ABOUT.

**Comment by Timberpoes** [comment]
Conflicting PR was closed #53602

**Pull request merged**